ui/hooks/version.ts

```diff
@@ -0,0 +1,25 @@
+import { useContext } from "react";
+import { useQuery } from "react-query";
+import { RequestError } from "../lib/types";
+import { GetVersionResponse } from "../lib/api/core/core.pb";
+import { CoreClientContext } from "../contexts/CoreClientContext";
+
+export type Version = {
+  version: string;
+  gitCommit: string;
+  branch: string;
+  buildTime: string;
+};
+
+export function useVersion() {
+  const { api } = useContext(CoreClientContext);
+
+  return useQuery<GetVersionResponse, RequestError>(
+    "version",
+    () => api.GetVersion({}),
+    {
+      staleTime: Infinity,
```

Comment:
Does this prevent `react-query` from re-fetching? Good to know.

Reply:
Yes, AFAIK, the query should never go stale (copied the options from the feature flag hook).